### PR TITLE
Refactoring continued

### PR DESCRIPTION
**Pull request by akallabeth**
* Improves logging
* Adds missing transportIO context
* Fixed uninitialized rdpSettings use

### CODE REVIEW

File: channels/drdynvc/client/drdynvc_main.c [hunk 1]
```diff
@@ -1510,9 +1511,12 @@ static UINT drdynvc_virtual_channel_event_connected(drdynvcPlugin* drdynvc, LPVO
 		return status;
 	}
 
-	settings = (rdpSettings*)drdynvc->channelEntryPoints.pExtendedData;
+	WINPR_UNUSED(drdynvc->rdpcontext);
+	settings = drdynvc->rdpcontext->settings;
```
Comment (MartinHaimberger) [inline channels/drdynvc/client/drdynvc_main.c:1515]: Copy past error, should it not be WINPR_ASSERT?

Comment (akallabeth) [reply channels/drdynvc/client/drdynvc_main.c:1515]: true.

File: libfreerdp/core/peer.c [hunk 2]
```diff
@@ -758,44 +779,62 @@ static int peer_recv_callback(rdpTransport* transport, wStream* s, void* extra)
...
+{
+	int rc = 0;
+	do
+	{
+		switch (rc)
+		{
+			default:
+				rc = peer_recv_callback_internal(transport, s, extra);
+				break;
+			case 2:
```
Comment (mfleisz) [inline libfreerdp/core/peer.c:831]: Not 100% sure but when I look at the following code in transport.c

```
		recv_status = transport->ReceiveCallback(transport, received, transport->ReceiveExtra);
		...
		/* session redirection or activation */
		if (recv_status == 1 || recv_status == 2)
```

this looping/rc handling might cause problems?

Comment (akallabeth) [reply libfreerdp/core/peer.c:831]: client side code, the other one is server side

File: libfreerdp/core/peer.c [hunk 3]
```diff
@@ -575,26 +580,29 @@ static int peer_recv_callback(rdpTransport* transport, wStream* s, void* extra)
...
 			else
 			{
-				IFCALLRET(client->Logon, client->authenticated, client, &client->identity, FALSE);
-			}
+				SelectedProtocol = nego_get_selected_protocol(rdp->nego);
+				settings->NlaSecurity = (SelectedProtocol & PROTOCOL_HYBRID) ? TRUE : FALSE;
+				settings->TlsSecurity = (SelectedProtocol & PROTOCOL_SSL) ? TRUE : FALSE;
+				settings->RdpSecurity = (SelectedProtocol == PROTOCOL_RDP) ? TRUE : FALSE;
 
+				if (SelectedProtocol & PROTOCOL_HYBRID)
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply libfreerdp/core/peer.c:591]: did not want to change everything, but you´re right.

**Pull request merged**